src/sync/tracks.nim

```diff
@@ -24,9 +24,6 @@ type
     tests*: TrackExerciseTests
     repoExercise: TrackRepoExercise
 
-proc newTrackRepo: TrackRepo =
-  result.dir = getCurrentDir()
-
```

Comment:
I made this change partly because I find it more readable to inline it in `findTrackExercises`. 

But there's also a general issue that I'll try to fix everywhere eventually: I believe that procs that initialize non-`ref` objects should be named starting with `init`, not `new` - even if they contain a field of a type like `string`. See https://nim-lang.github.io/Nim/nep1.html#introduction-naming-conventions

And in my refactoring branch I think I changed a lot of this stuff to use `distinct string` rather than an one-field object anyway.